woocommerce/grow
# Add GitHub action and workflow for the versioning management of the reusable custom actions

### Changes proposed in this Pull Request:

It's phase 4 of the migration plan in this comment - pb0Spc-2DL-p2#comment-5753.

- Create the github-actions JS package for the [GitHub JavaScript actions](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action).
- Add a new GitHub JavaScript action for versioning management.
   - [packages/js/github-actions/actions/update-version-tags](https://github.com/woocommerce/grow/tree/762083b0fd0195f925b67cf60463f302e7851ce9/packages/js/github-actions/actions/update-version-tags)
- Add a new GitHub workflow to build bundle and manage version tags of this repo when releasing the reusable custom actions.
   - [.github/workflows/github-actions-release.yml](https://github.com/woocommerce/grow/blob/762083b0fd0195f925b67cf60463f302e7851ce9/.github/workflows/github-actions-release.yml)

### Screenshots:

After releasing the `actions-v1.0.1-pre`, both references of `actions-v1-pre` and `actions-v1.0-pre` were updated onto the same reference of `actions-v1.0.1-pre`.

<img width="435" alt="2022-06-14 22 16 05" src="https://user-images.githubusercontent.com/17420811/173600223-47991b20-2776-49ee-834c-d8aa6dae7ce5.png">

### Detailed test instructions:

1. Go to the [release workflow](https://github.com/woocommerce/grow/actions/workflows/github-actions-release.yml) to review the release processing.
1. Go to the [1557721 commit](https://github.com/woocommerce/grow/commit/155772123dad4f5f2f0eb7d85d53bac50411c474) and the [1557721 revision](https://github.com/woocommerce/grow/tree/155772123dad4f5f2f0eb7d85d53bac50411c474) to view the released version `actions-v1.0.1-pre` built and committed via the release workflow.
1. Go to the [Releases](https://github.com/woocommerce/grow/releases) and [Tags](https://github.com/woocommerce/grow/tags) pages to check the results of versioning management.
1. Go to the Files changed and Check tabs of the RP 139-gh-woocommerce/automatewoo-referrals to see how the version tag `@actions-v1-pre` is used.

### Additional details:

About the versioning management, a more detailed description is documented [here](https://github.com/woocommerce/grow/tree/05fe94cbf256d43eee5ec920238446c3a5792842/packages/js/github-actions#official-release).

### Changelog entry

> Add - A new GitHub JavaScript action for the versioning management.
> Add - A new GitHub workflow for the bundle building and versioning management of the reusable custom actions in the repository when releasing.
